include/deal.II/hp/dof_handler.h

```diff
@@ -1090,12 +1106,16 @@ namespace hp
     std::vector<unsigned int> vertex_dof_offsets;
 
     /**
-     * Array to store the information if a cell on some level has children or
-     * not. It is used by the signal slots as a persistent buffer during the
-     * refinement, i.e. from between when pre_refinement_action is called and
-     * when post_refinement_action runs.
+     * Container to temporarily store the CellID and active FE index of
+     * cells that will be refined.
      */
-    std::vector<std::unique_ptr<std::vector<bool>>> has_children;
+    std::vector<std::pair<CellId, unsigned int>> refined_cells_fe_index;
```

Comment:
Why store the `CellId` instead of the cell iterator? You only use this for local and shared triangulations where the iterator of surviving cells remains valid after refinement/coarsening.